### PR TITLE
view service:  impl gas prices

### PR DESCRIPTION
**Pull request by Valentine1898**

### CODE REVIEW

File: packages/constants/config.ts [hunk 1]
```diff
@@ -8,6 +8,6 @@ export interface Config {
 
 export const testnetConstants: Config = {
   grpcEndpoint: 'https://grpc.testnet.penumbra.zone',
-  indexedDbVersion: 17,
+  indexedDbVersion: 18,
```
Comment (Valentine1898) [inline packages/constants/config.ts:11]: We must increment the indexed-db version every time we change indexed-db's structure

File: packages/query/src/block-processor/index.ts [hunk 2]
```diff
@@ -159,6 +159,7 @@ export class BlockProcessor implements BlockProcessorInterface {
       abortSignal: this.abortController.signal,
     })) {
       if (block.fmdParameters) await this.indexedDb.saveFmdParams(block.fmdParameters);
+      if (block.gasPrices) await this.indexedDb.saveGasPrices(block.gasPrices);
```
Comment (Valentine1898) [inline packages/query/src/block-processor/index.ts:162]: We get `GasPrices` from `CompactBlock` during the scanning process

**Review by jessepinho — APPROVED** [review]
Looks good to me! 

Just want to make sure I understand something correctly, though: it looks to me like we're setting gas prices in IndexedDB, detached from any one specific transaction. That is, any time we set the gas prices, they'll be set more or less "globally" in the `GAS_PRICES` object store.

Somehow this seems strange to me, although I'm having trouble identifying why... it's as if we're storing a transient value — that is, the current gas prices — in a non-transient store (IndexedDB). Then, each time we start a new transaction, we'll re-set the gas prices in IndexedDB. Wouldn't it be more appropriate to pass it around in memory while creating a new transaction, rather than storing it in IndexedDB? Or is that not possible, given the way our frontend architecture is structured (which I'm still learning about)? 

If there's a good reason for this, though, go ahead and merge! 🟢

**Review by TalDerei — APPROVED** [review]

**Comment by Valentine1898** [comment]
> Looks good to me!
> 
> Just want to make sure I understand something correctly, though: it looks to me like we're setting gas prices in IndexedDB, detached from any one specific transaction. That is, any time we set the gas prices, they'll be set more or less "globally" in the `GAS_PRICES` object store.
> 
> Somehow this seems strange to me, although I'm having trouble identifying why... it's as if we're storing a transient value — that is, the current gas prices — in a non-transient store (IndexedDB). Then, each time we start a new transaction, we'll re-set the gas prices in IndexedDB. Wouldn't it be more appropriate to pass it around in memory while creating a new transaction, rather than storing it in IndexedDB? Or is that not possible, given the way our frontend architecture is structured (which I'm still learning about)?
> 
> If there's a good reason for this, though, go ahead and merge! 🟢

Whenever Gas prices change they are published in CompactBlock. 
We do not know in which block Gas prices have changed, and we have no way to get the current Gas prices from penumbra full node. 
In practice it looks like this 
#0 - gas prices published in block 
#1 
...
#288
#289 - gas prices published in block 
#290
#291
...
#382 
#383 - user creates transaction and needs gas prices 
In block #383 we do not know in which block the actual value of gas prices is written. 
For this we need to cache the last known value of gas prices in indexed-db while we scan the blocks.

**Comment by jessepinho** [comment]
@Valentine1898 that explanation is suuuper helpful — thanks! ship it 🚢

**Comment by grod220** [comment]
That is super helpful documentation @Valentine1898. Low priority, but can you add that context to say here:

packages/router/src/grpc/view-protocol-server/gas-prices.ts

**Comment by Valentine1898** [comment]
> That is super helpful documentation @Valentine1898. Low priority, but can you add that context to say here:
> 
> packages/router/src/grpc/view-protocol-server/gas-prices.ts

Yeah, sure, I'll add

**Pull request merged**